Tasks/Xcode/xcode.ts

```diff
@@ -303,7 +303,10 @@ async function run() {
                         exportMethod = await sign.getProvisioningProfileType(embeddedProvProfiles[0]);
                         tl.debug('Using export method = ' + exportMethod);
                     }
-                    if (!exportMethod) {
+
+                    // If you create a simple macOS app with automatic signing and no entitlements, it won't have an embedded profile.
+                    // And export for that app will work with an empty exportOptionsPlist.
+                    if (!exportMethod && sdk === 'iphoneos') {
```

Comment:
We are relying on an optional input. Since we set it in the templates, it is fine for now. But seems like soon we will have to parse the project file.

Reply:
If `sdk` isn't set, but we were supplied a `xcWorkspace` argument, we could scan the project.pbxproj file for SDKROOT ("iphoneos", "macosx") and use that to drive which provisioning profile suffix we search for.